test_cb/test_cb_regression.py

```diff
@@ -114,9 +111,8 @@ def get_inputs_and_data_width(circuit):
                 tester.poke(getattr(genesis_cb, f"in_{i}"), inputs[i])
         tester.expect(genesis_cb.out, cb_functional_model(*_inputs))
         tester.eval()
-    testvectors = tester.test_vectors
 
     for cb in [genesis_cb, magma_cb]:
-        compile(f"test_cb/build/test_{cb.name}.cpp", cb, testvectors)
-        run_verilator_test(cb.name, f"test_{cb.name}", cb.name, ["-Wno-fatal"],
-                           build_dir="test_cb/build")
+        tester.circuit = cb
```

Comment:
it seems kind of hacky to overwrite `tester.circuit` so that we can test multiple circuits with the same interface. For now it's fine, but perhaps we could pass in a list of circuits which should all have the same `IO` and then call `compile_and_run(...)` with a specific one of the modules?

For now I think it's fine.

Reply:
Yea I agree, this was a convenient side effect of the current interface that I was abusing. I mainly chose this versus rerunning the test vector construction logic since that seemed like a bigger performance hit, although I don't think performance is really an issue for this test. We might consider separating out the notion of test vector construction from running tests. So, we could use the same test vector constructor for both circuits, and then run each test independently for each circuit.